Improves .editorconfig compatibility

pgmccann opened the pull request:
My text editor (NeoVim) reports the following error when editing a workshop template:

``` 
editorconfig: invalid value for option trim_trailing_whitespace: false  # keep trailing spaces in markdown - 2+ spaces are translated to a hard break (<br/>). trim_trailing_whitespace must be either "true" or "false"
editorconfig: invalid value for option max_line_length: 100  # please keep this in sync with bin/lesson_check.py!. max_line_length must be a number or "off"
```

As per https://editorconfig.org:

> Comments should go on their own lines.

This change brings .editorconfig in line with the file format as documented.

pgmccann committed:
Improves .editorconfig compatibility

My text editor (NeoVim) reports the following error when editing a workshop template:

``` 
editorconfig: invalid value for option trim_trailing_whitespace: false  # keep trailing spaces in markdown - 2+ spaces are translated to a hard break (<br/>). trim_trailing_whitespace must be either "true" or "false"
editorconfig: invalid value for option max_line_length: 100  # please keep this in sync with bin/lesson_check.py!. max_line_length must be a number or "off"
```

As per https://editorconfig.org:

> Comments should go on their own lines.

This change brings .editorconfig in line with the file format as documented.

pgmccann committed:
Merge branch 'carpentries:gh-pages' into gh-pages

dpshelio commented on .editorconfig:
```diff
@@ -8,8 +8,12 @@ trim_trailing_whitespace = true
+# Please keep this in sync with bin/lesson_check.py!
+
+trim_trailing_whitespace = false
+# keep trailing spaces in markdown - 2+ spaces are translated to a hard break (<br/>)
```
I think having the comments before the options are set (i.e., above rather than below) are clearer at signalling the reader what's coming and maybe safer that people would read it if changing it. But that may be a habit I've got.

pgmccann replied:
Thanks @dpshelio - I think you're right. I'll make that adjustment.

pgmccann committed:
Leading .editorconfig comments

Places .editorconfig comments before the corresponding commands rather than after.

tobyhodges reviewed (approved):
Thanks @pgmccann 🙌
